--- BUILD.bazel
+++ BUILD.bazel
@@ -1,4 +1,3 @@
-# Copyright 2017 Google Inc.
 # All Rights Reserved.
 #
 #
@@ -183,7 +182,9 @@
             "@fuchsia_sdk//pkg/zx",
         ],
         "//conditions:default": [],
-    }),
+    }) + ["@std_module//:std"],
+    module_interfaces = ["googletest.cppm"],
+    module_header_maps = ["googletest.modulemap"],
 )
 
 # `gtest`, but testonly. See guidance on `gtest` for when to use this.
